mode/python/python.js

```diff
@@ -249,15 +249,24 @@
     }
 
     function tokenLexer(stream, state) {
+      var sol = stream.sol();
+      if (sol){
+          state.whiteSpaceAtBeginningOfLine = true;
+      }
       var style = state.tokenize(stream, state);
       var current = stream.current();
 
       // Handle decorators
-      if (current == "@") {
-        if (parserConf.version && parseInt(parserConf.version, 10) == 3)
-          return stream.match(identifiers, false) ? "meta" : "operator";
-        else
-          return stream.match(identifiers, false) ? "meta" : ERRORCLASS;
+      if (state.whiteSpaceAtBeginningOfLine){
```

Comment:
You can just add this to the condition of the existing `if` with `&&`. No need to add another wrapping `if`